### PR TITLE
Proposals for more idiomatic use of delb

### PR DESCRIPTION
**Pull request by funkyfuture**
it looks rather less in the view here than i remember. maybe i messed up when i threw black over it. had forgotten do skip string normalization, hope that doesn't bother.

### CODE REVIEW

**Comment by JKatzwinkel** [comment]
nice, thanks!!!

File: peret/__init__.py [hunk 1]
```diff
@@ -121,7 +117,7 @@ def _has_translation(e: TagNode, lang: str, value: str) -> bool:
     for quote in e.css_select(
         f'entry > sense > cit[type="translation"][xml|lang="{lang}"] > quote'
     ):
-        if value in map(str, quote.child_nodes()):
+        if value in quote.full_text:
```
Comment (funkyfuture) [inline peret/__init__.py]: maybe this could also be a equality comparison.

Comment (JKatzwinkel) [reply peret/__init__.py]: it probably should, come to think of it

File: peret/__init__.py [hunk 2]
```diff
@@ -135,23 +131,13 @@ def _add_translation(e: TagNode, lang: str, value: str) -> TagNode:
...
-            children=[
-                e.new_tag_node(
-                    'quote',
-                    children=[value]
-                )
-            ]
+    if not e.css_select("entry > sense"):
+        e.append_child(tag("sense"))
+    e.css_select("entry > sense").first_child.append_child(
+        tag(
```
Comment (JKatzwinkel) [inline peret/__init__.py:136]: i love this

File: peret/__init__.py [hunk 3]
```diff
@@ -281,9 +259,7 @@ def _get_id(entry: TagNode) -> str:
 
     """
     # pylint: disable=protected-access
-    return _strip_id(
-        entry._etree_obj.xpath('@xml:id')[0]
-    )
+    return _strip_id(entry._etree_obj.xpath("@xml:id").first)
```
Comment (JKatzwinkel) [inline peret/__init__.py]: `.first` is not available here because i'm using `lxml` api directly...

File: peret/__init__.py (hunk 2 above)
Comment (JKatzwinkel) [reply peret/__init__.py:136]: the `tag` shortcut

File: peret/__init__.py [hunk 4]
```diff
@@ -135,23 +131,13 @@ def _add_translation(e: TagNode, lang: str, value: str) -> TagNode:
...
-            },
-            children=[
-                e.new_tag_node(
-                    'quote',
-                    children=[value]
-                )
-            ]
+    if not e.css_select("entry > sense"):
+        e.append_child(tag("sense"))
+    e.css_select("entry > sense").first_child.append_child(
```
Comment (JKatzwinkel) [inline peret/__init__.py]:
```suggestion
    e.css_select("entry > sense").first.append_child(
```

File: peret/__init__.py (hunk 3 above)
Comment (JKatzwinkel) [reply peret/__init__.py]:
```suggestion
    return _strip_id(entry._etree_obj.xpath("@xml:id")[0])
```

Comment (JKatzwinkel) [reply peret/__init__.py]: actually, how about this?
```suggestion
    return _strip_id(
        entry.attributes.get(f'{{{XML_NS}}}id')
    )
```

File: peret/__init__.py (hunk 1 above)
Comment (JKatzwinkel) [reply peret/__init__.py]:
```suggestion
    return e.css_select(
        f'entry > sense > cit[type="translation"][xml|lang="{lang}"] > quote'
    ).filtered_by(
        lambda quote: quote.full_text == value
    ).size > 0
```

**Review by JKatzwinkel — APPROVED** [review]

File: peret/__init__.py (hunk 1 above)
Comment (funkyfuture) [reply peret/__init__.py]: is this a one-liner competition? but yeah.

File: peret/__init__.py (hunk 3 above)
Comment (funkyfuture) [reply peret/__init__.py]: as a matter of fact, Python 3.9 eventually brought string methods to remove prefixes and suffixes.
for older ones also `"tla1234"[len("tla"):]` does it as well.

File: peret/__init__.py [hunk 5]
```diff
@@ -117,7 +117,11 @@ def _has_translation(e: TagNode, lang: str, value: str) -> bool:
     for quote in e.css_select(
         f'entry > sense > cit[type="translation"][xml|lang="{lang}"] > quote'
     ):
-        if value in quote.full_text:
+    return e.css_select(
+        f'entry > sense > cit[type="translation"][xml|lang="{lang}"] > quote'
+    ).filtered_by(
+        lambda quote: quote.full_text == value
+    ).size > 0
             return True
```
Comment (funkyfuture) [inline peret/__init__.py]:
```suggestion
        return e.css_select(
            f'entry > sense > cit[type="translation"][xml|lang="{lang}"] > quote'
        ).filtered_by(
            lambda quote: quote.full_text == value
        ).size > 0
```

Comment (funkyfuture) [inline peret/__init__.py]:
```suggestion
    return e.css_select(
        f'entry > sense > cit[type="translation"][xml|lang="{lang}"] > quote'
    ).filtered_by(
        lambda quote: quote.full_text == value
    ).size > 0
```

Comment (funkyfuture) [reply peret/__init__.py]: this?

Comment (funkyfuture) [reply peret/__init__.py]: dang it. go that at home.

Comment (funkyfuture) [reply peret/__init__.py]: at least i could break the interface. or vice versa.

Comment (JKatzwinkel) [reply peret/__init__.py]: yes please, that is what i meant!  i don't really get how multiline suggestions work on github.

**Pull request merged**